src/childinfo.c

```diff
@@ -136,7 +136,7 @@ void receiveChildInfo(void) {
 void receiveLastChildInfo(void) {
     if (server.child_info_pipe[0] == -1) return;
 
-    /* Drain the pipe and update child info */
+    /* Drain the pipe and update child info so that we get the final message. */
```

Comment:
Would be nice to refactor these two functions so the intent is clearer (since they're literally the same). It's probably worth merging this fix in though.

Reply:
i actually see no reason why we can't always use the one with the `while`